src/code/z_message_nes.c

```diff
@@ -196,14 +196,12 @@ void Message_LoadTimeNES(GlobalContext* globalCtx, u8 arg1, s32* offset, f32* ar
     u32 dayTime;
     s16 digits[4];
     f32 timeInMinutes;
-    s32 day;
     s16 i;
 
     if (arg1 == 0xCF) {
-        day = gSaveContext.save.day;
-        dayTime = 0x40000 - ((day % 5) << 16) - (u16)(-0x4000 + gSaveContext.save.time);
+        dayTime = (4 << 16) - (CURRENT_DAY << 16) - (u16)(-0x4000 + ((void)0, gSaveContext.save.time));
     } else {
-        dayTime = 0x10000 - (u16)(-0x4000 + gSaveContext.save.time);
+        dayTime = (1 << 16) - (u16)(-0x4000 + ((void)0, gSaveContext.save.time));
```

Comment:
Out of curiosity, what's the reason behind changing `0x40000` and `0x10000` to bit shifts?

Reply:
I interpreted as being part of the same term as `(CURRENT_DAY << 16)` i.e.:

`(DAY_4 << 16) - (CURRENT_DAY << 16) - remainder` -> `((DAY_4 - CURRENT_DAY) << 16) - remainder`

and in the else as an alternative:
`(DAY_1 << 16) - remainder`

I still don't entirely know what this time calculation is, but it seems to at the very least explain where those two numbers come from. Did you have a different interpretation on this?

Reply:
Well, actually from `EnBji01`, looks like this is getting the `timeBeforeMoonCrash` for the first line, and `timeBeforeEndOfDay` in the second line?

Reply:
Ah, I did not think of that. That makes sense. 
Since time is split up into 2 bytes, a full day would be represented in the `1 << 16` bit. So `((DAY_4 - CURRENT_DAY) << 16)` would be how many days until day 4. 
That also explains the `-0x4000` since time in this system is 24hr clock, but a new day starts at `6am` or `CLOCK_TIME(6, 0) = 0x4000` so it has to be offset.

Reply:
Nice! I might add a comment explaining this so it doesn't get lost

Reply:
Actually! What about something like this? This matches:

`timeLeft = (4 - CURRENT_DAY) * CLOCK_TIME(24, 0) - (u16)(((void)0, gSaveContext.save.time) - CLOCK_TIME(6, 0));`

Reply:
I like it. It seems clearer to me.